### ISSUE TITLE
Unable to use pass or drop with httpjson

### ISSUE THREAD

**Issue by aodj** [influxdata/telegraf#359]
As the title says, I can't seem to get any of the filtering mechanisms to work with the `httpjson` plugin (or the `exec` plugin fwiw)

In fact, adding in `pass` completely stops the plugin from working, even when using `-test` to validate:

```toml
# Read flattened metrics from one or more JSON HTTP endpoints
[httpjson]
  pass = [ "_all" ]
  # Specify services via an array of tables
  [[httpjson.services]]

    # a name for the service being polled
    name = "es_indexes"

    # URL of each server in the service's cluster
    servers = [
      "http://localhost:9200/testdata-*/_stats/docs/",
    ]

    # HTTP method to use (case-sensitive)
    method = "GET"

    # List of tag names to extract from top-level of JSON server response
    # tag_keys = [
    	# "_shards",
    # ]

    # HTTP parameters (all values must be strings)
    [httpjson.services.parameters]
      level = "cluster"
```
when executed as `telegraf -config httpjson.conf -test` returns simply:

    * Plugin: httpjson, Collection 1

... before exiting.

Basically what I'm trying to do is get data from an ElasticSearch cluster via it's [`Indices API`](https://www.elastic.co/guide/en/elasticsearch/reference/1.7/indices-stats.html). The data is returned as a JSON blob like so:

```json
{
  "_shards": {
    "total": 220,
    "successful": 220,
    "failed": 0
  },
  "_all": {
    "primaries": {
      "docs": {
        "count": 45341156,
        "deleted": 0
      }
    },
    "total": {
      "docs": {
        "count": 90682351,
        "deleted": 0
      }
    }
  }
}
```
This then gets dumped out by`httpjson` as the following:

```
* Plugin: httpjson, Collection 1
> httpjson_es_indexes__shards_total,server=http://localhost:9200/testdata-*/_stats/docs/ value=220 1447254066235166815
> httpjson_es_indexes__shards_successful,server=http://localhost:9200/testdata-*/_stats/docs/ value=220 1447254066235254165
> httpjson_es_indexes__shards_failed,server=http://localhost:9200/testdata-*/_stats/docs/ value=0 1447254066235284870
> httpjson_es_indexes__all_primaries_docs_count,server=http://localhost:9200/testdata-*/_stats/docs/ value=45341156 1447254066235314005
> httpjson_es_indexes__all_primaries_docs_deleted,server=http://localhost:9200/testdata-*/_stats/docs/ value=0 1447254066235339602
> httpjson_es_indexes__all_total_docs_count,server=http://localhost:9200/testdata-*/_stats/docs/ value=90682351 1447254066235367716
> httpjson_es_indexes__all_total_docs_deleted,server=http://localhost:9200/testdata-*/_stats/docs/ value=0 1447254066235392360
```
... of which I'm only interested in one line `httpjson_es_indexes__all_primaries_docs_count,server=http://localhost:9200/testdata-*/_stats/docs/ value=45341156 1447254066235314005`

It would be great if `httpjson` (and `exec` if you're feeling keen) could be made to support these since the documentation makes it appear that these so-called [**Plugin Options**](https://github.com/influxdb/telegraf#plugin-options) are actually supported by all plugins ("There are 5 configuration options that are configurable per plugin"...)

**Comment by aodj** [comment]
i need to correct my earlier assertion regarding `tagdrop` and `tagpass` since I had my syntax all wrong for it. 

```
[httpjson.tagdrop]
  server = [ "http://localhost:9200/testdata-*/_stats/docs/" ]
```
The above *will* filter out (since it's tagdrop) tags with the name `server` and value `http://localhost:9200/dohop-*/_stats/docs/`

**Comment by aodj** [comment]
After sleeping on it, I tried a brute force approach to work out what was happening; basically the value of the `pass` array need to start with the value of `name` if it's been specified. So in my case, I need to start all `pass` or `drop` filters with `es_indexes` which means I can get to the single value I want by specifying `pass = [ "es_indexes__all_primaries_docs_count" ]`

It's quite non-intuitive since you need to specify the`name` value as a filter for `pass` before you've even specified what it is.

**Comment by sparrc** [comment]
@aodj I'm not sure I understand, what do you want to be able to do? You want to do a glob-style matching of the pass filters?

Just inputing "_all" and expecting it to filter seems too broad, what if you have multiple metrics with "_all" but you only want some of them?

**Comment by sparrc** [comment]
With the current architecture there wouldn't really be a way around the "name" being part of the filter. In this case, "name" is an argument specific to this plugin, whereas "pass" is an argument for the entire daemon. The daemon doesn't have any control over the name of a measurement given by a plugin, it just takes whatever the plugin gives it and tries to filter.

I do think, though, that these filters should support glob matching. I'm going to make a case for that.

incidentally, there was a regression in the behavior of these filters, they actually also need to include the plugin name (see PR #360). I agree that this isn't ideal but the overall structure of measurements will be changing for Telegraf 0.3.0 (see #152)

**Comment by aodj** [comment]
@sparrc regarding your first post, `_all` was the name of a top level field in the JSON response from the API I was querying, rather than a glob syntax, since I was trying to work out the prefix required to filter on.

As for the second point you raise, I wasn't aware that the `pass` and `drop` fields were part of each plugin in turn, but rather thought they were attached to the individual items generating result sets ie, since it is the definition of `[[httpjson.services]]` that encapsulates and generates results, I thought it would also contain the `pass` and `drop` fields also.

Glob matching generally sounds useful so :+1: 

I noticed the regression you mention, and it actually threw me for a loop, since I upgraded from 0.1.9 to 0.2.0, and realised none of my telegraf instances were returning data.

Thanks for your response.
